gdx/src/com/badlogic/gdx/math/Matrix4.java

```diff
@@ -73,16 +73,19 @@ public class Matrix4 implements Serializable {
 	/** WW: Typically the value one. On Vector3 multiplication this value is ignored. */
 	public static final int M33 = 15;
 
-	static final Quaternion quat = new Quaternion();
-	static final Quaternion quat2 = new Quaternion();
-	static final Vector3 l_vez = new Vector3();
-	static final Vector3 l_vex = new Vector3();
-	static final Vector3 l_vey = new Vector3();
-	static final Vector3 tmpVec = new Vector3();
-	static final Matrix4 tmpMat = new Matrix4();
-	static final Vector3 right = new Vector3();
-	static final Vector3 tmpForward = new Vector3();
-	static final Vector3 tmpUp = new Vector3();
+	// avoid recursive constructors : stack overflow
+	private Matrix4 tmpMat = null;
```

Comment:
You could also change this to a thread-local, couldn't you?

Reply:
@dar-dev

Reply:
I think thread-local isn't strictly necessary here. This field is not _static_ so there's no issue when using different instances of Matrix4 per thread